flow/project.py

```diff
@@ -1508,6 +1513,75 @@ def add_operation_with_directives(function):
 
         return OperationRegister()
 
+    @staticmethod
+    def _setup_hook_object(parent_class):
+        class _HooksRegister:
+
+            _parent_class = parent_class
+
+            def __init__(self, hook_func, trigger):
+                """Add hooks to an operation workflow.
+
+                This object is designed to be used as a decorator. For example:
+
+                .. code-block:: python
+
+                    @FlowProject.operation
+                    @FlowProject.operation_hook.on_start(lambda op_name, job: print(op_name, job))
+                    def foo(job):
+                        pass
+
+                A hook is a function that is called at at specific points during the execution
+                of a job operation. In this example, the anonymous hook function
+                is executed before the operation **foo** runs. Hooks can also run after an
+                operation finish, when an operation exits with error,
+                or when an operation exists without error.
```

Comment:
Fixed some typos, improved wrapping consistency.
```suggestion
                A hook is a function that is called at at specific points
                during the execution of a job operation. In this example,
                the anonymous hook function is executed before the operation
                **foo** runs. Hooks can also run after an operation finishes,
                when an operation exits with error, or when an operation
                exits without error.
```

Reply:
Fixed. I edited the suggestion above. (Feel free to admin-edit my suggestions in the future for errors like that.)